backend/module/bot/slackbot/slackbot_helper.go

```diff
@@ -0,0 +1,13 @@
+package slackbot
+
+import (
+	"github.com/slack-go/slack"
+)
+
+func sendBotReply(client *slack.Client, channel, message string) error {
+	_, _, err := client.PostMessage(channel, slack.MsgOptionText(message, false))
```

Comment:
use the context versions of everything, e.g. `PostMessageContext` and thread it through the functions.

will save you a bunch of refactoring trouble later on.

Reply:
when threading the ctx through the slack client API returned `Post \"https://slack.com/api/chat.postMessage\": context canceled"}`. 

I think this is due to the asyc handling of the event / ack'ing the Events API - we return a response to the Events API before we've finished handling the event and so the original ctx gets cancelled . For a sanity check, I changed the flow to first handle the event and then ack the Events API, and I did not get this error.

`PostMessage` uses context.background by default https://github.com/slack-go/slack/blob/v0.8.2/chat.go#L123-L125

Reply:
yep, you should use a background context and not the request context when firing off asynchronous work in the background

Reply:
got it, thanks! updated